turn_tracker.gd

```diff
@@ -15,21 +19,59 @@ func _ready() -> void:
 	]
 	setTurnOrder(turnOrder)
 
+func _process(delta: float) -> void:
+	currentTurnTaker = Global.turnOrder[turnIndex]
 
 func resetTurnOrder() -> void:
 	Global.turnOrder = []
 
 func setTurnOrder(newOrder: Array) -> void:
 	Global.turnOrder = newOrder
 
+func resetNumberOfUnits() -> void:
+	numberOfUnits = 0
+
+func addUnits(count: int = 1) -> void:
+	numberOfUnits += count
+
 func subtractUnits(count: int) -> void:
 	numberOfUnits -= count
 
+func resetUnitsUsed() -> void:
+	unitsUsed = 0
+
 func addUnitsUsed(count: int = 1) -> void:
 	unitsUsed += count
 
+func subtractUnitsUsed(count: int = 1) -> void:
+	unitsUsed -= count
+
```

Comment:
could be useful if we add items that could undisable a unit and give them an extra turn? idk